prusa/link/web/files.py

```diff
@@ -84,34 +85,41 @@ def api_file_info(req, storage, path=None):
     file_tree = file.to_dict()
     result = file_tree.copy()
     file_type = result['type']
+    result['display_name'] = basename(path)
 
     # --- FOLDER ---
     # Fill children's tree data for the folder
     if file_type is FileType.FOLDER.value:
         for child in result.get("children", []):
+            child['display_name'] = child['name']
+            child_path = f'{path}/{child["name"]}'
+            child_os_path = f"{os_path}/{child['name']}"
+
             # Fill specific data for print files within children list
             if child["type"] is FileType.PRINT_FILE.value:
-                child_path = f'{path}/{child["name"]}'
-                child_os_path = f"{os_path}/{child['name']}"
                 child.update(fill_printfile_data(child_path, child_os_path,
-                                                 storage))
+                                                 storage, simple=True))
 
             # Fill specific data for firmware files within children list
-            # elif child["type"] is FileType.FIRMWARE.value:
+            elif child["type"] is FileType.FIRMWARE.value:
+                child.fill_file_data(child_path, storage)
 
             # Fill specific data for other files within children list
-            # elif child["type"] is FileType.FILE.value:
+            elif child["type"] is FileType.FILE.value:
+                child.fill_file_data(child_path, storage)
 
     # --- FILE ---
     # Fill specific data and metadata for print file
     elif file_type is FileType.PRINT_FILE.value:
         result.update(fill_printfile_data(path, os_path, storage))
 
     # Fill specific data for firmware file
-    # elif file_type is FileType.FIRMWARE.value:
+    elif file_type is FileType.FIRMWARE.value:
+        result.update(fill_file_data(path, storage))
 
     # Fill specific data for other file
-    # elif file_type is FileType.FILE.value:
+    elif file_type is FileType.FILE.value:
```

Comment:
What happen if is there another `file_type`? The better solution is expect that is `FileType.File` so use else here.

Reply:
Makes sense, I'll fix it.